code/game/gamemodes/wizard/spellbook.dm

```diff
@@ -383,9 +394,23 @@
 			user << "<span class='warning'>The contract has been used, you can't get your points back now!</span>"
 		else
 			user << "<span class='notice'>You feed the contract back into the spellbook, refunding your points.</span>"
-			src.uses++
+
+			uses++
+
+			for(var/datum/spellbook_entry/item/contract/CT in entries)
+				if(!isnull(CT.limit))
+					CT.limit++

```

Comment:
@TheDZD i might been just copying and pasting without thinking trying to get the vial limit in i will revert it back!